macros/plugins/spark/get_external_build_plan.sql

```diff
@@ -19,6 +19,13 @@
         {% set build_plan = build_plan + dbt_external_tables.refresh_external_table(source_node) %}
     {% endif %}
 
+    {% set recover_partitions = spark__recover_partitions(source_node) %}
+    {% if recover_partitions|length > 0 %}
+    {% set build_plan = build_plan + [
+        recover_partitions
+    ] %}
```

Comment:
> Users should run the `RECOVER PATITIONS` command for the source tables every time there is a new partition created.

Am I right to understand that `recover partitions` should only be run when I'm creating the table for the first time, or when I full-refresh it because I've changed the `external.partitions` config? If that's the case, I'd expect to see this in the `create_or_replace` build plan above.

Or is `recover partitions` needed _any_ time new data arrives, i.e. a new value is available for an existing partition? In which case, we should keep running it both when creating and partial-refreshing the external table.

Reply:
The later, therefor `RECOVER PARTITIONS` is needed for both full-refresh and partial-refreshing. 

To be precise: it is needed any time new data arrives that creates a _new_ partition. If new data arrives in an _existing_ partition, a refresh or cluster restart is enough.